src/main/java/trackr/model/task/TaskDeadline.java

```diff
@@ -0,0 +1,64 @@
+package trackr.model.task;
+
+import static java.util.Objects.requireNonNull;
+import static trackr.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a Task's deadline in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidTaskDeadline(LocalDate)}
+ */
+public class TaskDeadline {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Task deadline should only contain numeric values in the format \"DD/MM/YYYY\""
+                    + "and it should not be blank";
+
+    public final LocalDate taskDeadline;
+
+    /**
+     * Constructs a {@code TaskDeadline}.
+     *
+     * @param deadline A valid deadline.
+     */
+    public TaskDeadline(LocalDate deadline) {
+        requireNonNull(deadline);
+        checkArgument(isValidTaskDeadline(deadline), MESSAGE_CONSTRAINTS);
+        taskDeadline = deadline;
+    }
+
+    /**
+     * Returns true if a given local date is a valid deadline,
+     * meaning date is today's date or after today's date.
+     */
+    public static boolean isValidTaskDeadline(LocalDate test) {
+        return test.isAfter(LocalDate.now()) || test.isEqual(LocalDate.now());
+    }
+
+    /**
+     * Returns the deadline stored in "01 JANUARY 2023" format.
+     * @return A string representation of the deadline.
+     */
+    @Override
+    public String toString() {
+        int day = taskDeadline.getDayOfMonth();
+        String month = taskDeadline.getMonth().toString();
+        int year = taskDeadline.getYear();
+        return String.format("%d %s %d", day, month, year);
```

Comment:
Could be changed to perhaps
```suggestion
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd LLLL yyyy");
        return taskDeadline.format(formatter);
```
the formatter could be taken out and put as a static final variable outside of the function